tsl/src/data_node.c

```diff
@@ -541,22 +545,24 @@ add_distributed_id_to_data_node(TSConnection *conn)
 /*
  * Connect to do bootstrapping.
  *
- * This behaves similar to connectMaintenanceDatabase and will first try to
- * connect to "postgres" database and if that does not exists, to the
- * "template1" database.
+ * We iterate through the list of databases and try to connect to so we can
+ * bootstrap the data node.
  */
 static TSConnection *
 connect_for_bootstrapping(const char *node_name, const char *const host, int32 port,
 						  const char *username, const char *password)
 {
-	List *node_options = create_data_node_options(host, port, "postgres", username, password);
-	TSConnection *conn = remote_connection_open_with_options_nothrow(node_name, node_options);
-
-	if (conn)
-		return conn;
-
-	node_options = create_data_node_options(host, port, "template1", username, password);
-	return remote_connection_open_with_options_nothrow(node_name, node_options);
+	TSConnection *conn = NULL;
+	int i;
+	for (i = 0; i < MAX_BOOTSTRAP_DATABASES; i++)
+	{
+		List *node_options =
+			create_data_node_options(host, port, bootstrap_databases[i], username, password);
+		conn = remote_connection_open_with_options_nothrow(node_name, node_options);
+		if (conn)
+			return conn;
+	}
+	return conn;
```

Comment:
nit: maybe just return NULL here and possible to move `TSConnection *conn` to the loop block